app/src/main/java/org/wikipedia/descriptions/DescriptionEditFragment.java

```diff
@@ -312,7 +312,7 @@ public void twoFactorPrompt() {
         @SuppressWarnings("checkstyle:magicnumber")
         private void postDescription(@NonNull String editToken) {
 
-            disposables.add(ServiceFactory.get(pageTitle.getWikiSite()).getSiteInfo()
+            disposables.add(ServiceFactory.get(WikiSite.forLanguageCode(pageTitle.getWikiSite().languageCode())).getSiteInfo()
```

Comment:
Wouldn't this just result in pageTitle.getWikiSite() ?

Reply:
Nope! That's the key. When editing image captions, the value you get from `pageTitle.getWikiSite()` is commons.wikimedia.org, but the language code of the site is the true language, e.g. "en", "ru", etc.  The language is obviously not "commons".  So, when requesting `siteInfo` from commonswiki, we always get a language code of `en`, when in fact we want the `siteInfo` for the wiki that corresponds to the language code.